src/main/java/de/teamlapen/vampirism/blocks/TentBlock.java

```diff
@@ -146,11 +174,25 @@ public Direction getBedDirection(BlockState state, IWorldReader world, BlockPos
     }
 
     @Override
-    public void onReplaced(BlockState state, @Nonnull World world, @Nonnull BlockPos pos, @Nonnull BlockState newState, boolean isMoving) {
-        super.onReplaced(state, world, pos, newState, isMoving);
-        if (newState.getBlock() != state.getBlock()) {
-            forWholeTent(pos,state, (direction, blockpos) -> world.destroyBlock(blockpos,true));
-        }
+    public void onBlockHarvested(World world, @Nonnull BlockPos blockPos, BlockState blockState, @Nonnull PlayerEntity playerEntity) {
+        forWholeTent(blockPos,blockState, (direction, blockPos1) -> {
+            VampirismMod.dispatcher.sendToAllAround(new PlayEventPacket(1,blockPos1, Block.getStateId(world.getBlockState(blockPos1))),world.getDimension().getType(),(double)blockPos1.getX(), (double)blockPos1.getY(), (double)blockPos1.getZ(), 64);
+            world.destroyBlock(blockPos1, true);
+            spawnDrops(world.getBlockState(blockPos1), world, blockPos1, null, playerEntity, playerEntity.getHeldItemMainhand());
+        });
+    }
+
+    @OnlyIn(Dist.CLIENT)
+    public boolean hasCustomBreakingProgress(BlockState state) {
+        return true;
+    }
+
+    @Override
+    public void onReplaced(BlockState oldState, @Nonnull World world, @Nonnull BlockPos blockPos, @Nonnull BlockState newState, boolean p_196243_5_) {
```

Comment:
Is this necessary? onBlockHarvested should take care of this I assume. At least BlockBed does not use onReplaced